std-bits/database/src/main/java/org/enso/database/DatabaseConnectionDetailsSPI.java

```diff
@@ -27,6 +29,23 @@ public static String[][] get_default_constructors(boolean refresh) {
         .toArray(String[][]::new);
   }
 
+  /**
+   * Returns an array of all the types that implement the `DatabaseConnectionDetailsSPI` interface.
+   *
+   * @param refresh whether to refresh the list of types
+   * @return an array of all the types that implement the `DatabaseConnectionDetailsSPI` interface
+   */
+  public static Value[] get_types(boolean refresh) {
+    if (refresh) {
+      loader.reload();
+    }
+    return loader.stream().map(provider -> provider.get().getTypeObject()).toArray(Value[]::new);
```

Comment:
Does this iterate only through the SPI classes?

Reply:
Yes just iterates over the SPIs for the DatabaseConnectionDetails.

The loader specifies which ones it finds in the type arg.